spec/unit/pdk/telemetry_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+
+describe PDK::Telemetry do
+  let(:default_config) { {} }
+
+  before(:each) do
+    # We use a hard override to disable analytics for tests, but that obviously
+    # interferes with these tests...
+    ENV.delete('PDK_DISABLE_TELEMETRY')
+
+    # Ensure these tests will never read or write a local config
+    allow(described_class).to receive(:load_config).and_return(default_config)
+    allow(described_class).to receive(:write_config)
+  end
+
+  it 'creates a Noop client if telemetry is disabled' do
+    default_config.replace('disabled' => true)
+    expect(described_class).not_to receive(:write_config)
```

Comment:
Not sure you need this expectation.  This seems like an implementation detail, not a behaviour.

Reply:
Yep, those specs were just copy-pastad in from Bolt and hadn't been checked through yet.